src/commands/debug.js

```diff
@@ -0,0 +1,247 @@
+        w,
+        watch,
+      })
+    } catch (e) {
+      print.error(e.message)
+      process.exitCode = 1
+      return
+    }
+
+    // Fall back to default values for options / parameters
+    outputDir = outputDir && outputDir !== '' ? outputDir : filesystem.path('build')
+    manifest =
+      manifest !== undefined && manifest !== ''
+        ? manifest
+        : filesystem.resolve('subgraph.yaml')
+
+    // Show help text if requested
+    if (help) {
+      print.info(HELP)
+      return
+    }
+
+    // Validate the subgraph name
+    if (!subgraphName) {
+      print.error('No subgraph name provided')
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    // Validate endpoint
+    if (!endpoint) {
+      print.error(`No Graph node GraphQL endpoint provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    // Validate block
+    if (!block) {
+      print.error(`No block number provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    // Validate node
+    if (!node) {
+      print.error(`No Graph node provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    // Validate IPFS
+    if (!ipfs) {
+      print.error(`No IPFS node provided`)
+      print.info(HELP)
+      process.exitCode = 1
+      return
+    }
+
+    let protocol
+    try {
+      // Checks to make sure deploy doesn't run against
+      // older subgraphs (both apiVersion and graph-ts version).
+      //
+      // We don't want the deploy to run without these conditions
+      // because that would mean the CLI would try to compile code
+      // using the wrong AssemblyScript compiler.
+      await assertManifestApiVersion(manifest, '0.0.5')
+      await assertGraphTsVersion(path.dirname(manifest), '0.22.0')
+
+      const dataSourcesAndTemplates = await DataSourcesExtractor.fromFilePath(manifest)
+
+      protocol = Protocol.fromDataSources(dataSourcesAndTemplates)
+    } catch (e) {
+      print.error(e.message)
+      process.exitCode = 1
+      return
+    }
+
+    const isLocal = node.match(/localhost/) || node.match(/127\.0\.0\.1/)
+    if (!isLocal) {
+      print.error(`The Graph node must be running locally`)
+      process.exitCode = 1
+      return
+    }
+
+    let compiler = createCompiler(manifest, {
+      ipfs,
+      outputDir,
+      outputFormat: 'wasm',
+      skipMigrations: true,
+      blockIpfsMethods: false,  // Network does not support publishing subgraphs with IPFS methods
+      protocol,
+    })
+
+    // Exit with an error code if the compiler couldn't be created
+    if (!compiler) {
+      process.exitCode = 1
+      return
+    }
+
+    let requestUrl = new URL(node)
+    let client = createJsonRpcClient(requestUrl)
+
+    // Exit with an error code if the client couldn't be created
+    if (!client) {
+      process.exitCode = 1
+      return
+    }
+
+    let debugSubgraph = async ipfsHash => {
```

Comment:
I know the project isn't consistent in what uses `let` and `const`, but since this won't change it would be nice to be a constant 🙂

Reply:
Probably change it in [deploy.js](https://github.com/graphprotocol/graph-cli/blob/341eb940d8806f212b848a745dc8b1e101f5ef87/src/commands/deploy.js#L262) too?

Reply:
Also what do you think about `requestUrl`, `client`, `compiler`? Should we make them consts too? I don't know what the proper JS way is, but isn't const best suited for global vars, and declaring a local as const is a bit too verbose? Of course, there may be cases where using const for special values like arrow functions is beneficial.